packages/search-ui/src/SearchDriver.js

```diff
@@ -219,85 +222,147 @@ export default class SearchDriver {
       ...searchParameters
     };
 
+    // We want to make sure if this method is running, that any calls to this
+    // method that had been previously deferred are cancelled. We don't want to
+    // possibly have the deferred call execute after this immediate call.
+    this.debounceManager.cancelByName("_updateSearchResults");
```

Comment:
Is it worth noting in the comments exactly where the debounce for this is started? I had to search through the PR files to realize that it was initiated in setSearchTerm here:

https://github.com/elastic/search-ui/blob/126a3e6503e474a0d3e002cdbaf21249eee70ffc/packages/search-ui/src/actions/setSearchTerm.js#L32-L35

And to be fully honest, I'm still not 100% sure how setSearchTerm is inheriting `this._updateSearchResults` from SearchDriver 😖

Reply:
Yeah, that's a really good point.

And I'll tell you, but you're not going to like it... one sec.

Reply:
They're mixed in here:

https://github.com/elastic/search-ui/blob/master/packages/search-ui/src/SearchDriver.js#L98-L107

Reply:
Oh yeah, that's right, I totally forgot about that! It's been a while haha.

All good, but I think maybe just a short comment above with something like `this is initiated in the setSearchTerm action` would be helpful for developer navigation.